primedev/mods/modmanager.cpp

```diff
@@ -619,6 +623,53 @@ void ModManager::SearchFilesystemForMods()
 	std::sort(m_LoadedMods.begin(), m_LoadedMods.end(), [](Mod& a, Mod& b) { return a.LoadPriority < b.LoadPriority; });
 }
 
+void ModManager::DisableMultipleModVersions()
+{
+	// Stores versions, for each mod, associated to their position in the `m_LoadedMods` array, *e.g.*:
+	//
+	// {
+	//     "Northstar.Client": [ {"1.30.2", 0} ],
+	//     "Northstar.Custom": [ {"1.30.2", 1} ],
+	//     "Northstar.CustomServers": [ {"1.30.2", 2} ],
+	//     "Extraction": [ {"1.2.0", 3}, {"1.2.1", 4}, {"1.3.0", 5} ]
+	// }
+	//
+	std::unordered_map<std::string, std::vector<std::tuple<const char*, int>>> modVersions;
+
+	// Load up the dictionary
+	int i = 0;
+	for (Mod& mod : m_LoadedMods)
+	{
+		// Store versions for enabled mods only, as disabled mods are not loaded and won't collide
+		if (!mod.m_bEnabled)
+		{
+			continue;
+		}
+
+		modVersions[mod.Name].push_back({mod.Version.c_str(), i});
+		i++;
+	}
+
+	// Find duplicate mods and disable them
+	for (const auto& pair : modVersions)
+	{
+		if (pair.second.size() <= 1)
+		{
+			continue;
+		}
+
+		spdlog::warn("Mod '{}' has several versions enabled, disabling them all.", pair.first);
+		for (const std::tuple<std::string, int> tVersion : pair.second)
+		{
+			std::string version = std::get<std::string>(tVersion);
+			int versionIndex = std::get<int>(tVersion);
```

Comment:
```suggestion
		for (auto &[version, versionIndex] : pair.second)
		{
```
Structured bindings are cool (but if you don't like them, you need to change the tuple type here, I think you forgot)

No guarantees this compiles, github web editor things. Might need to make it not a reference or something if it doesn't like it

Reply:
You're the cpp expert here, I take your advice with pleasure :smiley:
Tested locally, seems to work as intended.